Assets/js/customer.js

```diff
@@ -249,4 +252,73 @@ document.getElementsByClassName("filter--div")[0].addEventListener('click', func
         initialProducts.sort(compare);
         showCarts();
     }
-}); 
\ No newline at end of file
+}); 
+
+
+// Search bar
+// Name Array
+let nameArray = [];
+initialProducts.map(function(cartItem,i){
+    if(!(nameArray.includes(cartItem.productName))){
+        nameArray.push(cartItem.productName);
+    }
+});
+
+// create li and a elements
+for(let i=0; i<nameArray.length; i++){
+    let li = document.createElement('li');
+    li.className = "li-search";
+    document.getElementById("myUL").appendChild(li);
+    let a = document.createElement('a');
+    a.className = "a-search";
+    a.innerText = nameArray[i];
+    li.appendChild(a);
+}
+// Search Suggestions
+document.getElementById("search-input").addEventListener("keyup", function(){
```

Comment:
You could also listen to the `"change"` event instead of the `"keyup"` event, and that will capture some more edge cases.  For example, when the user right clicks on the input field and pastes something, the `"change"` event will capture that but the `"keyup"` event won't.

Reply:
I changed it, but it took along time for search suggestions to appear

Reply:
Sorry, I told you the wrong event name.  I meant to say `"input"` event, not `"change"` event.
The difference is that "input" fires whenever the value changes, but "change" only fires when the value is committed (for example after the user presses enter).
Sorry about that! If you are still interested, you could see if "input" fixes the slow search suggestions.

Reply:
I tried it, and yes it worked. 
thanks for the clarification.

Reply:
🎉